sandbox/plc4c/api/include/plc4c/subscribe.h

```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef PLC4C_SUBSCRIBE_H_
+#define PLC4C_SUBSCRIBE_H_
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "plc4c/types.h"
+
+/*
+ *
+ * Subscriptions
+ *
+ */
+
+/**
+ * Destroys a given subscription-request.
+ *
+ * @param subscription_request the subscription-request.
+ */
+void plc4c_subscription_request_destroy(
+    plc4c_subscription_request *subscription_request);
+
+/**
+ * Returns the plc4c_connection for a give subscription request
+ * @param subscription_request plc4c_subscription_request
+ * @return plc4c_connection
+ */
+plc4c_connection *plc4c_subscription_request_get_connection(
+    plc4c_subscription_request *subscription_request);
+
+/**
+ * Sets the plc4c_connection for a given subscription request
+ * @param subscription_request plc4c_subscription_request
+ * @param connection plc4c_connection
+ */
+void plc4c_subscription_request_set_connection(
+    plc4c_subscription_request *subscription_request,
+    plc4c_connection *connection);
+
```

Comment:
I don't understand the difference between these three things.  Can you explain them?

Reply:
Cyclic item, Event Item, Change of state item